source/guides/first-steps/task-script.md

```diff
@@ -1,4 +1,125 @@
+
+The player is found by their username with the `<server.match_player[]>` tag, which returns a PlayerTag of the player whose username matches the text you put between the `[]` brackets. If this is confusing, don't worry! We'll talk more about tags later on - this is just an explanatory preview to give you an idea of what's coming. You can also return to it later on once you understand more about these concepts.
+
+# Building your first task script in VS Code
+
+Previously, you've learned how to create an environment in VS Code, the editor that we recommend for writing Denizen scripts.
+
+## Creating the file
+
+To create your first task script, start by opening your scripts folder in VS Code.
+
+TKTK SCREENSHOT (open scripts folder)
+
+From there, right click the scripts folder in the explorer menu and click the "New File" option.
+
+TKTK SCREENSHOT (new file highlighted)
+
+Once your file has been created, you can name it something like `MyTask.dsc`. Make sure to use the `.dsc` extension so VS Code recognizes it as a Denizen script, but the *name* of the file can be whatever you like. Usually, it's helpful to name the file something relevant to the script(s) that it contains.
+
+From here, double-click the file to open it in VS Code.
+
+TKTK SCREENSHOT (opened file)
+
+Now you can begin writing your first task script!
+
+## Writing the script
+
+Let's start with the core of the script:
+
+```dscript_blue
+myfirsttask:
+    type: task
+    script:
+    - narrate (sometext)
+```
+
+This example should look familiar - it's very similar to the example above.
+
+### Script names
+
+The name of the script here is `myfirsttask`. It's at the top of the indentation. Each top-level indentation entry is a separate script. Here is an example of two different scripts:
+
+```dscript_green
+myfirsttask:
+    type: task
+    script:
+    - narrate "This is is task number one!"
+
+mysecondtask:
+    type: task
+    script:
+    - narratee "This is task number two!"
+```
+
+This example demonstrates what having two different scripts in the same file looks like. `myfirsttask` and `mysecondtask` are two different scripts, and each is a fully self-contained script that can be run with the `run` command, either via the `/ex` command in-game or in a Denizen script.
+
+### Script types
+
+Below the script name - on the second level of indentation - you'll see the `type` and `script` entries.
+
+The `type` option is where you specify which `type` of Denizen script you're writing. In this case, we're writing a `task` script. You may have seen `world` scripts, `item` scripts, `inventory` scripts, and *maybe* even a `procedure` script. For now, let's focus on `task` scripts. Make sure you write `type: task` in your script on the second level of indentation, below the script name, like this:
```

Comment:
If you're going to start listing script types like that - maybe mention as well that there's a large variety of types (outside of the few examples listed) each of which serves a different purpose